src/api/l_audio.c

```diff
@@ -119,21 +119,40 @@ static int l_lovrAudioCapture(lua_State* L) {
   return 1;
 }
 
-static int l_lovrAudioSetCaptureDevice(lua_State *L) {
-  //
-
-  return 0;
-}
-
-static int l_lovrAudioGetCaptureDevice(lua_State *L) {
-  //
+static int l_lovrAudioGetDevices(lua_State *L) {
+  AudioDevice *devices;
+  size_t count;
+  lovrAudioGetDevices(&devices, &count);
+  lua_newtable(L);
+  int listOfDevicesIdx = lua_gettop(L);
+  for(int i = 0; i < count; i++) {
+    AudioDevice *device = &devices[i];
+    lua_pushinteger(L, i+1); // key in listOfDevicesIdx, for the bottom settable
+    lua_newtable(L);
+    luax_pushenum(L, AudioType, device->type);
+    lua_setfield(L, -2, "type");
+    lua_pushstring(L, device->name);
+    lua_setfield(L, -2, "name");
+    lua_pushboolean(L, device->isDefault);
+    lua_setfield(L, -2, "isDefault");
+    lua_pushlightuserdata(L, device->identifier);
+    lua_setfield(L, -2, "identifier");
+    lua_pushinteger(L, device->minChannels);
+    lua_setfield(L, -2, "minChannels");
+    lua_pushinteger(L, device->maxChannels);
+    lua_setfield(L, -2, "maxChannels");
+
+    lua_settable(L, listOfDevicesIdx);
```

Comment:
I procrastinated doing this fix for WEEKS because... lua's stack thinking really does my head in. But now I think I got it. Fix applied. Hope I got it right.